frontend/src/components/organisms/AddingPlayerForm/AddingPlayerForm.tsx

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import { Grid, Typography } from "@material-ui/core";
+import Controls from "../../atoms/Shared/controls/Controls";
+import { VerticalSpace } from "../../atoms/Shared/VerticalSpace";
+import { useForm, Form } from "../useForm/useForm";
+
+const initialFormValues = {
+  name: "",
+  surname: "",
+  email: "",
+  phone: "",
+};
+
+function AddingPlayerForm() {
+  const validate = (fieldValues = values) => {
+    const temp = {
+      name: "",
+      surname: "",
+      email: "",
+      phone: "",
+    };
+    if ("name" in fieldValues) {
+      temp.name =
+        fieldValues.name.trim() !== "" ? "" : "To pole jest wymagane.";
+    }
+    if ("surname" in fieldValues) {
+      temp.surname =
+        fieldValues.surname.trim() !== "" ? "" : "To pole jest wymagane.";
+    }
+    if ("email" in fieldValues) {
+      temp.email = /\S+@\S+\.\S+/.test(fieldValues.email)
+        ? ""
+        : "Niepoprawny adres e-mail.";
+    }
+    if ("phone" in fieldValues) {
+      temp.phone =
+        fieldValues.phone.trim().length >= 9
+          ? ""
+          : "Numer telefonu powinien zawierać przynajmniej 9 znaków";
+    }
+    setErrors({ ...temp });
+
+    if (fieldValues === values) {
+      return Object.values(temp).every((value: string) => value === "");
+    }
+  };
+
+  const { values, setValues, errors, setErrors, handleInputChange } = useForm(
+    initialFormValues,
+    true,
+    validate
+  );
```

Comment:
Mega wygląda

Reply:
Będę musiał się wzorować na Tobie, jeśli chodzi o formularze

Reply:
no teraz tworząc nowy formularz używamy tego useForm'a właśnie, a nie, że piszemy jakiś od nowa... a pola jeżeli mamy same TextInputy --> Inputy u nas to wszystko jest gotowe, tylko sobie robisz osobny komponent na formularz i wpisujesz odpowiednie labelki... a jeśli chcesz nowe inputy (radiobuttony, check'i jakieś, datePicker) to dodajesz do Control nowy komponent z tym i wrzucasz do formularza - polecam ten godzinny filmik, tam wszystko jest ładnie wyjaśnione. 
Kto będzie brał się za jakiś formularz to fajnie, żeby obczaił ten filmik, bo będzie wiedział jak użyć tego co zrobiłem - a jeszcze przynajmniej dwa formularze zostały nam do zrobienia

Reply:
Super, to mega jeśli działa tak jak powinno! Właśnie miało to tak chodzić, niezła robota!